pkg/azurefile/controllerserver.go

```diff
@@ -853,32 +853,67 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
+	// 	}
+
+	// 	tp := timestamppb.New(properties.LastModified())
+	// 	if tp == nil {
+	// 		return nil, status.Errorf(codes.Internal, "Failed to convert timestamp(%v)", properties.LastModified())
+	// 	}
+
+	// 	createResp := &csi.CreateSnapshotResponse{
+	// 		Snapshot: &csi.Snapshot{
+	// 			SizeBytes:      volumehelper.GiBToBytes(int64(properties.Quota())),
+	// 			SnapshotId:     sourceVolumeID + "#" + snapshotShare.Snapshot(),
+	// 			SourceVolumeId: sourceVolumeID,
+	// 			CreationTime:   tp,
+	// 			// Since the snapshot of azurefile has no field of ReadyToUse, here ReadyToUse is always set to true.
+	// 			ReadyToUse: true,
+	// 		},
+	// 	}
+
+	// 	isOperationSucceeded = true
+	// 	return createResp, nil
+	// } else {
+	snapshotShare, err := d.cloud.FileClient.WithSubscriptionID(subsID).CreateFileShare(ctx, rgName, accountName, &fileclient.ShareOptions{Name: fileShareName, RequestGiB: defaultAzureFileQuota, Metadata: map[string]*string{snapshotNameKey: &snapshotName}}, "snapshots")
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to get share url with (%s): %v", sourceVolumeID, err)
+		return nil, status.Errorf(codes.Internal, "create snapshot from(%s) failed with %v, accountName: %q", sourceVolumeID, err, accountName)
 	}
 
-	snapshotShare, err := shareURL.CreateSnapshot(ctx, azfile.Metadata{snapshotNameKey: snapshotName})
+	klog.V(2).Infof("Created share snapshot: %s", snapshotShare.Header.Get("x-ms-snapshot"))
+
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "create snapshot from(%s) failed with %v, shareURL: %q", sourceVolumeID, err, shareURL)
+		return nil, status.Errorf(codes.Internal, "failed to get snapshot properties from (%s): %v", snapshotShare.Header.Get("x-ms-snapshot"), err)
 	}
 
-	klog.V(2).Infof("Created share snapshot: %s", snapshotShare.Snapshot())
-
-	properties, err := shareURL.GetProperties(ctx)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to get snapshot properties from (%s): %v", snapshotShare.Snapshot(), err)
+	if snapshotShare.SnapshotTime == nil {
+		return nil, status.Errorf(codes.Internal, "Last modified time of snapshot is null")
 	}
 
-	tp := timestamppb.New(properties.LastModified())
+	tp := timestamppb.New(snapshotShare.SnapshotTime.Time)
 	if tp == nil {
-		return nil, status.Errorf(codes.Internal, "Failed to convert timestamp(%v)", properties.LastModified())
+		return nil, status.Errorf(codes.Internal, "Failed to convert timestamp(%v)", snapshotShare.SnapshotTime.Time)
 	}
 
 	createResp := &csi.CreateSnapshotResponse{
 		Snapshot: &csi.Snapshot{
-			SizeBytes:      volumehelper.GiBToBytes(int64(properties.Quota())),
-			SnapshotId:     sourceVolumeID + "#" + snapshotShare.Snapshot(),
+			SizeBytes:      volumehelper.GiBToBytes(int64(*snapshotShare.ShareQuota)),
+			SnapshotId:     sourceVolumeID + "#" + snapshotShare.Header.Get("x-ms-snapshot"),
```

Comment:
`snapshotShare.Header.Get("x-ms-snapshot")` it's always empty, use `snapshotShare.SnapshotTime.Time` instead.